packages/shared_preferences/shared_preferences_platform_interface/lib/shared_preferences_platform_interface.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import 'method_channel_shared_preferences.dart';
+
+/// The interface that implementations of shared_preferences must implement.
+///
+/// Platform implementations should extend this class rather than implement it as `shared_preferences`
+/// does not consider newly added methods to be breaking changes. Extending this class
+/// (using `extends`) ensures that the subclass will get the default implementation, while
+/// platform implementations that `implements` this interface will be broken by newly added
+/// [SharedPreferencesStorePlatform] methods.
+abstract class SharedPreferencesStorePlatform {
+  /// The default instance of [SharedPreferencesStorePlatform] to use.
+  ///
+  /// Defaults to [MethodChannelSharedPreferencesStore].
+  static SharedPreferencesStorePlatform get instance => _instance;
+
+  /// Platform-specific plugins should set this with their own platform-specific
+  /// class that extends [SharedPreferencesStorePlatform] when they register themselves.
+  static set instance(SharedPreferencesStorePlatform value) {
+    try {
+      instance._verifyProvidesDefaultImplementations();
+      _instance = value;
+    } on NoSuchMethodError catch (_) {}
+  }
+
+  static SharedPreferencesStorePlatform _instance =
+      MethodChannelSharedPreferencesStore();
+
+  /// Only mock implementations should set this to true.
+  ///
+  /// Mockito mocks are implementing this class with `implements` which is forbidden for anything
+  /// other than mocks (see class docs). This property provides a backdoor for mockito mocks to
+  /// skip the verification that the class isn't implemented with `implements`.
+  @visibleForTesting
+  bool get isMock => false;
+
+  /// Removes the value associated with the [key].
+  Future<bool> remove(String key);
+
+  /// Stores the [value] associated with the [key].
+  ///
+  /// The [valueType] must match the type of [value] as follows:
+  ///
+  /// * Value type "Bool" must be passed if the value is of type `bool`.
+  /// * Value type "Double" must be passed if the value is of type `double`.
+  /// * Value type "Int" must be passed if the value is of type `int`.
+  /// * Value type "String" must be passed if the value is of type `String`.
+  /// * Value type "StringList" must be passed if the value is of type `List<String>`.
+  Future<bool> setValue(String valueType, String key, Object value);
+
+  /// This does not do anything.
+  ///
+  /// It is only here for backwards-compatibility.
```

Comment:
I found this comment confusing. This is the 1.0 version of the platform interface, so there's no need to be backwards compatible. Can we just remove it instead of baking this flaw into the platform API forever? Deprecating/removing `commit` from the app-facing API can be done later.

Reply:
Currently the plugin sends a `commit` message over to the Java/Objective-C side via a platform channel. I don't know how breaking it would be to stop sending this message, but I was thinking of adding Web support without breaking changes. However, if you are OK with it I'll remove it.

Reply:
I just checked the iOS and Android implementations, and they just return `true` when they get a `commit` call. I think we should remove it from the platform interface, and when you migrate the plugin, just have `commit` return `true` directly.

Reply:
Done.